src/int.cr

```diff
@@ -357,8 +357,46 @@ struct Int
     (self & mask) == mask
   end
 
-  def gcd(other : Int)
-    self == 0 ? other.abs : (other % self).gcd(self)
+  # Returns the greatest common divisor of `self` and `other`. Signed
+  # integers may raise overflow if either has value equal to `MIN` of
+  # its type.
+  #
+  # ```
+  # 5.gcd(10) # => 2
+  # 5.gcd(7)  # => 1
+  # ```
+  def gcd(other : self) : self
+    u = self.abs
+    v = other.abs
+    shift = self.class.zero
```

Comment:
This line can be moved beneath two `return` lines below.

Reply:
:+1: